core/types/tx_setcode.go

```diff
@@ -0,0 +1,238 @@
+// address.
+type Authorization struct {
+	ChainID uint64         `json:"chainId" gencodec:"required"`
+	Address common.Address `json:"address" gencodec:"required"`
+	Nonce   uint64         `json:"nonce" gencodec:"required"`
+	V       uint8          `json:"v" gencodec:"required"`
+	R       *big.Int       `json:"r" gencodec:"required"`
+	S       *big.Int       `json:"s" gencodec:"required"`
+}
+
+// field type overrides for gencodec
+type authorizationMarshaling struct {
+	ChainID hexutil.Uint64
+	Nonce   hexutil.Uint64
+	V       hexutil.Uint64
+	R       *hexutil.Big
+	S       *hexutil.Big
+}
+
+// SignAuth signs the provided authorization.
+func SignAuth(auth *Authorization, prv *ecdsa.PrivateKey) (*Authorization, error) {
+	h := prefixedRlpHash(
+		0x05,
+		[]interface{}{
+			auth.ChainID,
+			auth.Address,
+			auth.Nonce,
+		})
+
+	sig, err := crypto.Sign(h[:], prv)
+	if err != nil {
+		return nil, err
+	}
+	return auth.withSignature(sig), nil
+}
+
+// withSignature updates the signature of an Authorization to be equal the
+// decoded signature provided in sig.
+func (a *Authorization) withSignature(sig []byte) *Authorization {
+	r, s, _ := decodeSignature(sig)
+	cpy := Authorization{
+		ChainID: a.ChainID,
+		Address: a.Address,
+		Nonce:   a.Nonce,
+		V:       sig[64],
+		R:       r,
+		S:       s,
+	}
+	return &cpy
+}
+
+type AuthorizationList []*Authorization
+
+// Authority recovers the authorizing
+func (a Authorization) Authority() (common.Address, error) {
+	sighash := prefixedRlpHash(
+		0x05,
+		[]interface{}{
+			a.ChainID,
+			a.Address,
+			a.Nonce,
+		})
+	if !crypto.ValidateSignatureValues(a.V, a.R, a.S, true) {
+		return common.Address{}, ErrInvalidSig
+	}
+	// encode the signature in uncompressed format
+	r, s := a.R.Bytes(), a.S.Bytes()
+	sig := make([]byte, crypto.SignatureLength)
+	copy(sig[32-len(r):32], r)
+	copy(sig[64-len(s):64], s)
+	sig[64] = a.V
+	// recover the public key from the signature
+	pub, err := crypto.Ecrecover(sighash[:], sig)
+	if err != nil {
+		return common.Address{}, err
+	}
+	if len(pub) == 0 || pub[0] != 4 {
+		return common.Address{}, errors.New("invalid public key")
+	}
+	var addr common.Address
+	copy(addr[:], crypto.Keccak256(pub[1:])[12:])
+	return addr, nil
+}
+
+// copy creates a deep copy of the transaction data and initializes all fields.
+func (tx *SetCodeTx) copy() TxData {
+	cpy := &SetCodeTx{
+		Nonce: tx.Nonce,
+		To:    tx.To,
+		Data:  common.CopyBytes(tx.Data),
+		Gas:   tx.Gas,
+		// These are copied below.
+		AccessList: make(AccessList, len(tx.AccessList)),
+		AuthList:   make(AuthorizationList, len(tx.AuthList)),
+		Value:      new(uint256.Int),
+		ChainID:    tx.ChainID,
+		GasTipCap:  new(uint256.Int),
+		GasFeeCap:  new(uint256.Int),
+		V:          new(uint256.Int),
+		R:          new(uint256.Int),
+		S:          new(uint256.Int),
+	}
+	copy(cpy.AccessList, tx.AccessList)
+	copy(cpy.AuthList, tx.AuthList)
```

Comment:
![스크린샷 2024-12-02 오후 2 57 14](https://github.com/user-attachments/assets/e929e41e-7405-4720-be66-f92fc5d9b66b)
Since AuthList is in pointer format, I think deep copying should be used here, not shallow copying. (This issue does not directly impact)

`type AuthorizationList []*Authorization`

Reply:
Good point. I have changed the authorization to a list of values.